src/libraries/Common/src/Interop/Windows/BCrypt/Interop.BCryptEncryptDecrypt.RSA.cs

```diff
@@ -117,6 +117,13 @@ private enum BCryptEncryptFlags : uint
                     dwFlags);
             }
 
+            // Windows 10.1903 can return success when it meant NTE_BUFFER_TOO_SMALL.
+            if (status == NTSTATUS.STATUS_SUCCESS && written > destination.Length)
+            {
+                bytesWritten = 0;
```

Comment:
It looks like when this occurs it does a partial write, so I think it should `ZeroMemory` on `destination` to not leave partially decrypted data in the buffer.

Reply:
If that's the case then we would be already missing that in the NCrypt version. Should that be addressed too?

Reply:
Looks like we do it here for NCrypt? 

https://github.com/bartonjs/corefx/blob/841660bd978b3259556d8d5abf320c39b2d33278/src/Common/src/System/Security/Cryptography/RSACng.EncryptDecrypt.cs#L242-L244

The `TryEncryptOrDecrypt` just ends up calling `EncryptOrDecrypt`.

Reply:
Alright, so it's done somewhere...

I was looking at this code path:
https://github.com/dotnet/runtime/blob/14cc72586a50ebed83da10988b19b8b31f22fda3/src/libraries/Common/src/System/Security/Cryptography/RSACng.EncryptDecrypt.cs#L254-L279